core/src/Kokkos_View.hpp

```diff
@@ -174,9 +186,11 @@ class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
   // FIXME: these are overriden so that their types are identical when using
   // BasicView or Legacy we will need to obtain these from base_t in the future
   // and deprecate old behavior
-  using size_type    = typename memory_space::size_type;
-  using value_type   = typename traits::value_type;
-  using pointer_type = typename traits::value_type*;
+  using size_type  = typename memory_space::size_type;
+  using value_type = typename traits::value_type;
+  // pointer_type can be different from element_type*
+  using pointer_type = decltype(Impl::ptr_from_data_handle(
+      std::declval<typename base_t::data_handle_type>()));
```

Comment:
You don't want to use a qualified call to `ptr_from_data_handle` here but rather rely on ADL, right?

Reply:
yes

Reply:
If we ever needed this we could make it ADL, but right now this would allow you to introduce ADL calls without going through any explicit opt-in mechanism like the Kokkos::(Impl::)ViewArguments. So I wouldn't like this design this way.

Reply:
In that case, the `static_assert` is misleading. For a custom data handle you would need to provide an overload in namespace `Kokos::Impl` suggesting this isn't allowed in the first place. Hence, I would just remove the text from the `static_assert`.